javascript/javascript1/week2/candy.js

```diff
@@ -0,0 +1,36 @@
+const boughtCandyPrices = [];
+const candyTablePrice = [
+  ["sweet", 0.5],
+  ["chocolate", 0.7],
+  ["toffee", 1.1],
+  ["chewing-gum", 0.03],
+];
+
+function addCandy(candyType, weight) {
+  for (i = 0; i < candyTablePrice.length; i++) {
+    {
+      if (candyTablePrice[i][0] === candyType) {
+        let priceActual = candyTablePrice[i][1] * weight;
+        return priceActual;
+      }
+    }
+  }
+}
```

Comment:
This for-loop, can be avoided. But first, good job on learning how to use objects. Next:

If you create you `candyTablePrice` as follows:
```javascript
const candyTablePrice = {
  sweet: 0.5,
  chocolate: 0.7,
  toffee: 1.1,
  chewinggum: 0.03,
};
```

then you can acces the price like:
```javascript
const candyName = "sweet"
console.log(candyTablePrice[candyName])
```

meaning the whole for loop can be replaced with this lookup instead of iterating over all the elements including the equal operator

Clear?

Reply:
Yes, That was much cleaner code and efficient one.